File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int getMinimumDifference(TreeNode* root) {
        vector<int> v;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            TreeNode* cur = q.front();
            q.pop();
            if(!cur) continue;
            v.push_back(cur->val);
            q.push(cur->left);
            q.push(cur->right);
        }
        sort(v.begin(), v.end());
        int m = INT_MAX;
        for(int i=0;i<v.size()-1;i++){
            m = min(m, abs(v[i] - v[i+1]));
        }
        return m;
    }
};

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int min = Integer.MAX_VALUE;
    TreeNode prev = null;
    public int getMinimumDifference(TreeNode root) {
        List<Integer> list = new ArrayList<>();
        find(root);
        return min;
    }
    private void find(TreeNode node){
        if(node == null) return;
        find(node.left);
        if(prev != null){
            min = Math.min(min, Math.abs(node.val - prev.val));
        }
        prev = node;
        find(node.right);
    }
}